pkg/genericapiserver/api/handlers/responsewriters/status.go

```diff
@@ -49,20 +48,14 @@ func apiStatus(err error) *metav1.Status {
 		//TODO: check for invalid responses
 		return &status
 	default:
-		status := http.StatusInternalServerError
-		switch {
-		//TODO: replace me with NewConflictErr
-		case storage.IsTestFailed(err):
```

Comment:
Read the comment below: "these typically indicate programmer by REST storage - these typically indicate programmer error". And in case of a programming error we tweak the error message to not look as ugly as a programming error.